### PR TITLE
feat(lambda): Lambda Getting Started project

### PR DESCRIPTION
**Pull request by seshubaws**
## Problem
It is difficult for new users to get started using their local IDE to build Lambda applications.

## Solution
Implemented new section "Application Builder" that adds quick action icons to build, locally invoke, and deploy/sync. Also added a guided walkthrough to offer starter templates and quick installation of prereqs.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -31,6 +47,3926 @@
         "./notifications": "./dist/src/notifications/index.js"
     },
     "contributes": {
+        "configuration": {
```
Comment (justinmk3) [inline packages/core/package.json:50]: this doesn't look right. this stuff now lives in `packages/toolkit/package.json` , not the core.

File: packages/core/src/awsService/appBuilder/activation.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import { getDeployWizard, runDeploy } from '../../shared/sam/deploy'
+import { DeployTypeWizard } from './wizards/deployTypeWizard'
+
+export const templateToOpenAppComposer = 'aws.toolkit.appComposer.templateToOpenOnStart'
+
+/**
+ * Activates the AWS Explorer UI and related functionality.
+ *
+ * IMPORTANT: Views that should work in all vscode environments (node or web)
+ * should be setup in {@link activateViewsShared}.
```
Comment (justinmk3) [inline packages/core/src/awsService/appBuilder/activation.ts:33]: was this comment just copy-paste, or is it actually relevant here?

File: packages/core/src/awsService/appBuilder/explorer/detectSamProjects.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { SamAppLocation } from './samProject'
+import { getLogger } from '../../../shared/logger/logger'
+import { getProjectRootUri } from '../../../shared/sam/utils'
+
+export async function detectSamProjects(): Promise<SamAppLocation[]> {
```
Comment (justinmk3) [inline packages/core/src/awsService/appBuilder/explorer/detectSamProjects.ts:11]: why is this file in /explorer/ ? seems relevant in the /sam/ family of modules.

Comment (seshubaws) [reply packages/core/src/awsService/appBuilder/explorer/detectSamProjects.ts:11]: It's used exclusively to populate the app builder view if a SAM project was detected in the workspace so I believe that's why we have it here. We can track moving this as a fast follow after launch.

File: packages/core/src/shared/utilities/downloadPatterns.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    const response = await fetch(url)
+    if (!response.ok) {
+        getLogger().error(`Failed to fetch the latest release: ${response.statusText}`)
+        throw new ToolkitError(`Failed to fetch the latest release: ${response.statusText}`)
+    }
+    return Buffer.from(await response.arrayBuffer())
+}
+
+async function unzipPattern(buffer: Buffer, outputDir: vscode.Uri, removeTopDir: boolean): Promise<void> {
```
Comment (justinmk3) [inline packages/core/src/shared/utilities/downloadPatterns.ts:26]: what does "pattern" refer to here? needs a docstring

File: packages/toolkit/.changes/next-release/Feature-336a5c62-92cc-4e73-a375-d63eaddb9cc6.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "AWS Lambda: AWS Explorer Command Pallette actions and hover buttons to build, local invoke, deploy, and remote invoke"
```
Comment (justinmk3) [inline packages/toolkit/.changes/next-release/Feature-336a5c62-92cc-4e73-a375-d63eaddb9cc6.json:3]:
```suggestion
	"description": "AWS Lambda: AWS Explorer command palette actions and hover buttons to build, local invoke, deploy, and remote invoke"
```

**Review by justinmk3 — APPROVED** [review]
the core/package.json comment is a blocker

File: packages/core/package.json (hunk 1 above)
Comment (seshubaws) [reply packages/core/package.json]: corrected this, thank you for catching!

**Pull request merged**